r-package/grf/R/print.R

```diff
@@ -81,3 +90,37 @@ print.boosted_regression_forest <- function(x, ...) {
     cat("Boosted GRF object", "\n")
     cat("Number of forests: ",length(x$forests), "\n")
 }
+
+
+print_tuning_params <- function(tuning.output, nq, p) {
+  # Print average error for nq-quantiles of tuned parameters
+  # Extra branches for mtry and min.node.size (e.g. cannot form quintiles
+  # for mtry if the number of variables is less than 5)
+  grid = tuning.output$grid
+
+  out = lapply(colnames(grid)[-1], function(name) {
+    m = nq
+    if (name == "mtry")
+      m = min(p - 1, nq)
+    probs = 0:m/m
+    q = quantile(grid[, name], probs = probs)
+    if (length(unique(q) < length(q)))
+      q = unique(q)
+    rank = cut(grid[, name], q, include.lowest=TRUE)
+    out = aggregate(grid[, "error"], by=list(rank), FUN=mean)
+    colnames(out) = c(name, "error")
+    out
+  })
+
+  err = tuning.output$error
+  params = tuning.output$params[colnames(grid)[-1]]
+  opt = formatC(c(err, params))
+
+  cat("Optimal tuning parameters: \n")
+  cat(paste0(names(opt), ": ", opt, "\n"))
+
+  cat("Average error by ", nq, "-quantile:\n", sep="")
+  for (i in out) {
+    print(i, row.names = FALSE)
```

Comment:
Small comment, I think it'd be more readable to have a space between each section.